### PR TITLE
feat: delete documents using delete worker

### PR DESCRIPTION
**Pull request by christyjacob4**
## What does this PR do?
In this PR, we leverage the delete worker to delete dangling documents when a collection is deleted. 

## Test Plan
While there is no concrete way to test if the documents are deleted today, we will keep this as a technical debt and revisit this later. The issue can be tracked here #752

E2E tests have been added to test delete collection. 

## Related PRs and Issues
#521 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
Yes

### CODE REVIEW

File: app/workers/deletes.php [hunk 1]
```diff
@@ -10,31 +10,43 @@
...
 
     public function perform()
     {
+        $projectId = $this->args['projectId'];
         $document = $this->args['document'];
+
         $document = new Document($document);
         
-        switch ($document->getCollection()) {
+        switch (strval($document->getCollection())) {
```
Comment (eldadfux) [inline app/workers/deletes.php:35]: You can also use:

```php
(string)$document->getCollection()
```

Just syntax sugar, don't have to change it.

File: app/workers/deletes.php [hunk 2]
```diff
@@ -10,31 +10,43 @@
...
                 break;
+            case Database::SYSTEM_COLLECTION_FUNCTIONS:
+                $this->deleteFunction($document, $projectId);
+                break;
             case Database::SYSTEM_COLLECTION_USERS:
-                $this->deleteUser($document);
+                $this->deleteUser($document, $projectId);
+                break;
+            case Database::SYSTEM_COLLECTION_COLLECTIONS:
+                $this->deleteDocuments($document,$projectId);
```
Comment (eldadfux) [inline app/workers/deletes.php:46]: Missing space after `,`

**Review by eldadfux — CHANGES_REQUESTED** [review]
Minor fix and a note :)

File: app/workers/deletes.php (hunk 1 above)
Comment (christyjacob4) [reply app/workers/deletes.php:35]: Sure @eldadfux

File: app/workers/deletes.php (hunk 2 above)
Comment (christyjacob4) [reply app/workers/deletes.php]: Done 👍

**Comment by christyjacob4** [comment]
Is it okay to add #521 as a related issue for this?? Or do we have a dedicated issue to track this ?

**Comment by eldadfux** [comment]
#521 should be great for this.

**Comment by christyjacob4** [comment]
Done 👍

File: app/workers/deletes.php (hunk 2 above)
Comment (eldadfux) [reply app/workers/deletes.php]: Still don't see the updated space 🤔

Comment (eldadfux) [reply app/workers/deletes.php]: Seen it! sorry!

**Review by eldadfux — CHANGES_REQUESTED** [review]

File: tests/e2e/Services/Database/DatabaseCustomServerTest.php [hunk 3]
```diff
@@ -5,10 +5,121 @@
...
     use ProjectCustom;
     use SideServer;
+
+    public function testDeleteCollection()
+    {
+        /**
+         * Test for SUCCESS
+         */
+
+        var_dump("Test Starting");
```
[the remark this answers is not in the document]
Comment (christyjacob4) [reply tests/e2e/Services/Database/DatabaseCustomServerTest.php]: Done

**Review by eldadfux — APPROVED** [review]

**Pull request merged**